cluster-autoscaler/clusterstate/clusterstate.go

```diff
@@ -893,3 +893,13 @@ func getNotRegisteredNodes(allNodes []*apiv1.Node, cloudProvider cloudprovider.C
 	}
 	return notRegistered, nil
 }
+
+// GetClusterSize calculates and returns cluster's current size and target size. The current size is the
+// actual number of nodes registered in Kubernetes, the target size is the number of nodes the CA wants.
+func (csr *ClusterStateRegistry) GetClusterSize() (currentSize, targetSize int) {
+	for _, accRange := range csr.acceptableRanges {
```

Comment:
Is it thread safe?

Reply:
Let's add locking here.

BTW you shouldn't really trust in thread safety of clusterstate. With exception of RegisterScaleDown nothing is ever called from a different goroutine and nothing is really tested for thread safety. Actually, we should probably revisit this at some point. I'll open an issue to do so.